lib/DigestSender.php

```diff
@@ -0,0 +1,215 @@
+		IConfig $config,
+		Data $data,
+		UserSettings $userSettings,
+		GroupHelper $groupHelper,
+		IMailer $mailer,
+		IUserManager $userManager,
+		IURLGenerator $urlGenerator,
+		Defaults $defaults,
+		IFactory $l10nFactory,
+		IManager $activityManager,
+		IDateTimeFormatter $dateTimeFormatter
+	) {
+		$this->config = $config;
+		$this->data = $data;
+		$this->userSettings = $userSettings;
+		$this->groupHelper = $groupHelper;
+		$this->mailer = $mailer;
+		$this->userManager = $userManager;
+		$this->urlGenerator = $urlGenerator;
+		$this->defaults = $defaults;
+		$this->l10nFactory = $l10nFactory;
+		$this->activityManager = $activityManager;
+		$this->dateFormatter = $dateTimeFormatter;
+	}
+
+	public function sendDigests(int $now) {
+		$users = $this->getDigestUsers();
+		$userLanguages = $this->config->getUserValueForUsers('core', 'lang', $users);
+		$userTimezones = $this->config->getUserValueForUsers('core', 'timezone', $users);
+		$defaultLanguage = $this->config->getSystemValue('default_language', 'en');
+		$defaultTimeZone = date_default_timezone_get();
+
+		foreach ($users as $user) {
+			$language = (!empty($userLanguages[$user])) ? $userLanguages[$user] : $defaultLanguage;
+			$timezone = (!empty($userTimezones[$user])) ? $userTimezones[$user] : $defaultTimeZone;
+			$this->sendDigestForUser($user, $now, $timezone, $language);
+		}
+	}
+
+	/**
+	 * get all users who have activity digest enabled
+	 *
+	 * @return string[]
+	 */
+	private function getDigestUsers(): array {
+		return $this->config->getUsersForUserValue('activity', 'notify_setting_activity_digest', 1);
+	}
+
+	private function getLastSendActivity(string $user, int $now): int {
+		$lastSend = (int)$this->config->getUserValue($user, 'activity', 'activity_digest_last_send', 0);
+		if ($lastSend > 0) {
+			return $lastSend;
+		}
+
+		return $this->data->getFirstActivitySince($user, $now - (24 * 60 * 60));
+	}
+
+	public function sendDigestForUser(string $uid, int $now, string $timezone, string $language) {
+		$l10n = $this->l10nFactory->get('activity', $language);
+		$lastSend = $this->getLastSendActivity($uid, $now);
+		$user = $this->userManager->get($uid);
+		if ($lastSend === 0) {
+			return;
+		}
+
+		['count' => $count, 'max' => $lastActivityId] = $this->data->getActivitySince($uid, $lastSend);
+		if ($count == 0) {
+			return;
+		}
+
+		/** @var IEvent[] $activities */
+		$activities = $this->data->get(
+			$this->groupHelper,
+			$this->userSettings,
+			$uid,
+			$lastSend,
+			self::ACTIVITY_LIMIT,
+			'asc',
+			'all',
+			'',
+			0,
+			true
+		);
+		$skippedCount = max(0, $count - self::ACTIVITY_LIMIT);
+
+		$template = $this->mailer->createEMailTemplate('activity.Notification', [
+			'displayname' => $user->getDisplayName(),
+			'url' => $this->urlGenerator->getAbsoluteURL('/'),
+			'activityEvents' => $activities,
+			'skippedCount' => $skippedCount,
+		]);
+		$template->setSubject($l10n->t('Daily activity digest for ' . $this->defaults->getName()));
```

Comment:
Better wording:
> Daily Nextcloud activity summary

Just to be clear – this is about activity, not only notifications, right? Otherwise it should say "Daily Nextcloud notification summary" of course.

Reply:
> this is about activity, not only notifications

yes